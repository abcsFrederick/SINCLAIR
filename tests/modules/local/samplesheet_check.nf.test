nextflow_process {

    name "Test Process SAMPLESHEET_CHECK"
    script "modules/local/samplesheet_check.nf"
    process "SAMPLESHEET_CHECK"

    test("samplesheet_check snapshot test") {

        when {
            params {
            }
            process {
                """
                input[0] = file(params.input)
                input[1] = file(params.contrast)
                input[2] = params.run_cellranger
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("samplesheet_check snapshot test - INF in contrastsheet") {

        when {
            params {
            }
            process {
                """
                input[0] = file(params.contrast_test.input)
                input[1] = file(params.contrast_test.contrast)
                input[2] = params.contrast_test.run_cellranger
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
