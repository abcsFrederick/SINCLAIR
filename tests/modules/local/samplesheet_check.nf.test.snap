{
    "samplesheet_check snapshot test": {
        "content": [
            {
                "0": [
                    "project_gex_samplesheet.csv:md5,da57547681b84c09b1eb173cae014a53"
                ],
                "1": [

                ],
                "2": [
                    "project_contrast_samplesheet.csv:md5,a11ed7f973f579f771ab1ac48effd315"
                ],
                "3": [
                    "project_groups_samplesheet.csv:md5,b4643b0b4bc7bf076f1d4c9ee2008a39"
                ],
                "atac_samplesheet": [

                ],
                "contrast_samplesheet": [
                    "project_contrast_samplesheet.csv:md5,a11ed7f973f579f771ab1ac48effd315"
                ],
                "gex_samplesheet": [
                    "project_gex_samplesheet.csv:md5,da57547681b84c09b1eb173cae014a53"
                ],
                "group_samplesheet": [
                    "project_groups_samplesheet.csv:md5,b4643b0b4bc7bf076f1d4c9ee2008a39"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-04-16T15:38:52.646431"
    }
}
